src/Numerics/LinearRegression/DemingRegression.cs

```diff
@@ -0,0 +1,203 @@
+        /// Computations are offset to be centered around (mean(x), mean(y)),
+        /// for improved numerical stability.
+        /// </remarks>
+        public static (double A, double B, double C) Fit(double[] x, double[] y, double delta = 1.0)
+        {
+            if (x.Length != y.Length)
+            {
+                throw new ArgumentException($"All sample vectors must have the same length. However, vectors with disagreeing length {x.Length} and {y.Length} have been provided. A sample with index i is given by the value at index i of each provided vector.");
+            }
+
+            if (x.Length <= 1)
+            {
+                throw new ArgumentException($"A regression of the requested order requires at least {2} samples. Only {x.Length} samples have been provided.");
+            }
+
+            // First Pass: Mean and abs-max values
+            double mx = 0.0;
+            double my = 0.0;
+            double maxax = 0;
+            double maxay = 0;
+            for (int i = 0; i < x.Length; i++)
+            {
+                mx += x[i];
+                my += y[i];
+
+                double ax = Math.Abs(x[i]);
+                double ay = Math.Abs(y[i]);
+                if (ax > maxax)
+                    maxax = ax;
+                if (ay > maxay)
+                    maxay = ay;
+            }
+
+            // Mean values
+            mx /= x.Length;
+            my /= y.Length;
+
+            // Second Pass: Second-degree central sample moments
+            double sxx = 0.0, sxy = 0.0, syy = 0;
+            for (int i = 0; i < x.Length; i++)
+            {
+                double xdiff = x[i] - mx;
+                double ydiff = y[i] - my;
+                sxx += xdiff * xdiff;
+                sxy += xdiff * ydiff;
+                syy += ydiff * ydiff;
+            }
+
+            sxx /= x.Length;
+            sxy /= x.Length;
+            syy /= x.Length;
+
+            // Need to either solve for a for y = -a*x:
+            //    −sxy * a^2 + (syy-delta*sxx) * a + delta*sxy = 0
+            // or solve for b for x = -b*y:
+            //    −sxy * b^2 + (sxx-syy/delta) * b + sxy/delta = 0
+            // Which to choose: Depends on whether the slope is larger than one
+
+            // If sxy is zero, either:
+            //   - data is all symmetric around the mid-point
+            //   - x or y is constant
+            // Then either a or b, or both, are zero
+            if (Math.Abs(sxy) <= (delta * sxx + syy) * 1e-10)
+            {
+                // When sxx has a value, it is a vertical line
+                if (delta * sxx > syy && sxx > maxax * 1e-10)
+                    return (0, 1, -my);
+                // when syy has a value, it is a horizontal line
+                else if (syy > maxay * 1e-10)
+                    return (1, 0, -mx);
+                // No line could be calculated
+                else
+                    return (0, 0, 0);
```

Comment:
I would expect an algorithm which fails to return `double.NaN`s instead of zeros